test/test_python_package_change_task.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+import os
+import unittest
+
+from pipeline.tasks import python_grpc_tasks
+
+
+class PythonPackageChangeTest(unittest.TestCase):
+
+    _TASK = python_grpc_tasks.PythonChangePackageTask()
+
+    _PROTO_FILE = [
+        '# Comment line\n',
+        'package google.service.v1;\n',
+        'import "google/service/a.proto";\n',
+        'import "google/cloud/otherapi/b.proto";\n',
+        'import "google/common/common_proto.proto";\n',
+        'Some other text referencing to google.service.v1\n']
+
+    def test__extract_base_dirs(self):
+        mock_proto = mock.mock_open()
+        mock_proto.return_value.__iter__ = lambda _: iter(self._PROTO_FILE)
+        with mock.patch('__builtin__.open', mock_proto, create=True):
+            base_dirs = self._TASK._extract_base_dirs(
+                os.path.join('a', 'test', 'path', 'to', 'google', 'service',
+                             'v1', 'a.proto'))
+            repeated_base_dirs = self._TASK._extract_base_dirs(
+                os.path.join('home', 'foo', 'workspaces', 'google', 'service',
+                             'v1', 'googleapis', 'google', 'service', 'v1',
+                             'a.proto'))
```

Comment:
Oops, sorry I suggested a wrong test case. What about `/home/foo/workspaces/google/some/v1/googleapis/google/service/v1/a.proto`?  I want to make sure it skips `google/some/v1` part.